abi/dynamic/reflect_marshal_test.go

```diff
@@ -67,6 +65,30 @@ func TestDynamicMarshal(t *testing.T) {
 	}
 }
 
+func TestDynamicMarshalErrors(t *testing.T) {
+	require := require.New(t)
+
+	abiJSON := mustReadFile(t, "../testdata/abi.json")
+	var abi abi.ABI
+
+	err := json.Unmarshal(abiJSON, &abi)
+	require.NoError(err)
+
+	t.Run("malformed JSON", func(t *testing.T) {
+		malformedJSON := `{"uint8": 42, "uint16": 1000, "uint32": 100000, "uint64": 10000000000, "int8": -42, "int16": -1000, "int32": -100000, "int64": -10000000000,`
+		_, err := Marshal(abi, "MockObjectAllNumbers", malformedJSON)
+		require.Error(err)
```

Comment:
Yes, `require := require.New(t)` is ugly. I'm going to keep it as is though, since it requires jumping through a few hoops to make the proper require available in the subtest.